### DISCUSSION TITLE
Have slinger working with 2 slingboxes

### DISCUSSION THREAD

**Discussion by mdehrlich** [GerryDazoo/Slinger#85] [General]
Thank you for your efforts.  Struggled a bit, but learned a lot.  I have a 350 and an M1.  Using 2 config files with one sling.server.  Config is an old Raspberry PI B+  I cannot believe how well this works.  I use an Moesgo remote.  It is awesome.  I can control my cable, DVD and amp with it from an app on my phone.  Only $20.  I have a slight problem with my 350 connected to a FIOS arris vms1100.  If TV is turned off, the output of the arris seems to jump to 1080p.  The slingbox cannot handle this format.  Hoping to fix with a display emulator passthough plug. If this doesn't work, i may buy a 500 so I can use HDMI.

**Comment by GerryDazoo** [comment]
I believe you can fix the output resolution on the arris. I suspect you have it set to Auto. ON my box on the front panel there's an output resolution button that will cycle through the valid outputs.

**Comment by mdehrlich** [comment]
Don't see a device button for resolution.  The DVR (software) settings are 1080i, not auto.  This works fine while the tv is powered on.  With the tv off, the HDMI output jumps to 1080p.  This messes up the slingbox connected by component video.  In the next few days I should get delivery of this pass-though adapter that says it fakes the video source into believing that the TV is on, and at the same resolution as when it was on.

**Comment by theweatherguy** [comment]
Hi  mdehrlich,
We currently run two slingboxes in our house on the same network.  
I'd like to do the same when the regular service is discontinued in November- run two slingboxes on the same slinger server and same network.  

How did you accomplish this? You said you used two separate config files.  Did you rename one of them to be config2.ini?    What changes were needed to be made to reference the second config in the python server script?     Thank you in advance for any help.

**Reply by mdehrlich** [reply]
Yes, create another copy of config.ini I named my config files M1_config.ini and 350_config.ini  Then add the names of both config files to the end of line 11 in sling.service
Make sure you use another port number in the second config file.  I used 8082.  It's that simple.

**Reply by theweatherguy** [reply]
Thank you very much.  Sounds easy.   But I opened  the **sling.service** file and I'm not sure where or how I would place my two config file names in that file.    It doesn't look like the sling.service has any reference to config.ini on any line.

Any specific help would be appreciated.

Here's sling.service—
[Unit]
Description=SlingBox Server Service
After=multi-user.target

[Service]
Type=idle
User=slingbox
WorkingDirectory=/home/slingbox
StandardOutput=append:/tmp/sling.log
StandardError=append:/tmp/sling.log
ExecStart=/usr/bin/python -u slingbox_server.py
Restart=always

[Install]
WantedBy=multi-user.target

**Reply by mdehrlich** [reply]
add the configs to this line
ExecStart=/usr/bin/python -u slingbox_server.py  config1.ini config2.ini

The reason you don't see config.ini is that the program defaults to config.ini if you don't add the filenames.

**Reply by theweatherguy** [reply]
Thank you! thank you!

**Reply by theweatherguy** [reply]
I renamed config.ini to config1.ini
I created a new config2.ini for our second slingbox.
I added the two config files to the sling.service line 11.   ExecStart=/usr/bin/python -u slingbox_server.py config1.ini config2.ini

I got an error message below
What did I do wrong?


_slingbox$ python3 slingbox_server.py
Version : 3.07 Running on macOS-12.6-x86_64-i386-64bit pid= 1481
Can't find specified config file config.ini Giving up.
Exception in thread Thread-1 (ConnectionManager):
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/threading.py", line 1016, in _bootstrap_inner
Traceback (most recent call last):
  File "/Users/slingbox/slingbox_server.py", line 1100, in <module>
    serverinfo = cp['SERVER']
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/configparser.py", line 964, in __getitem__
    raise KeyError(key)
KeyError: 'SERVER'
    self.run()
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/threading.py", line 953, in run
    self._target(*self._args, **self._kwargs)
  File "/Users/slingbox/slingbox_server.py", line 823, in ConnectionManager
    serverinfo = cp['SERVER']
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/configparser.py", line 964, in __getitem__
    raise KeyError(key)
KeyError: 'SERVER'_

**Reply by GerryDazoo** [reply]
Your confusing two things: running the program as a service and running it manually.

Manually:
 python3 slingbox_server.py config1.ini config2.ini

As a service:
    I'm NOT sure how to do this on a mac but something like
sudo systemd restart sling.service 

PS. Did you make sure to change [SERVER] port in your second config to something other than 8080? 8082 will work.

**Reply by theweatherguy** [reply]
I'm probably confusing more than two things!  :-)
I will work on this later.   Thanks for your help.   As for my other 'hat' as theweatherguy.  That was quite a storm you had in Nova Scotia.  As a barometer fan, it was impressive.   Glad to hear you're OK.   Visited Nova Scotia for a bike tour a few years ago.  Loved it.

**Reply by theweatherguy** [reply]
One more thing to clarify.   So you're saying I don't modify the sling.service file.  Instead I use the command:

_python3 slingbox_server.py config1.ini config2.ini_    

when I manually start the program in Terminal? And each config file should have a separate/different port, like 8080 and 8082

**Reply by theweatherguy** [reply]
That worked!!  So I didn’t modify the sling.service file. I used the command in Terminal—

_python3 slingbox_server.py config1.ini config2.ini_

when starting the slinger server.  

Only thing I noticed with two sling boxes.  It takes my iPad player much more time to buffer and display the video.

Next project to get this working outside our home with port forwarding or my wireguard vpn.

Thanks again for everyone's generous help.

**Reply by theweatherguy** [reply]
It works with my home network Wireguard VPN server without port forwarding!  Both slingboxes accessed using VPN on iPhone with cellular service and wireless turned temporarily off. Using VLC player on the iPhone 

I’m putting together a summary for other Mac users of how to get it going and I’ll post when it’s completed. 

Thank you, JerryDazoo, for your  help and brilliance in putting this together and the others here that have been so helpful!

**Comment by mdehrlich** [comment]
My primary cable box  (FIOS Arris VMS1100) would change to 60p whenever the TV is turned off.  The 350 slingbox cannot do 60p, so I would get a " Input not supported" in the slingplayer.  With VLC I get a blank screen.  I tried lots of pass though dongles from Amazon, but none worked.  I finally found one that does work.  The cable box thinks it has an always on 60i TV, so it stays in 60i mode.  The unit costs less than $16.  https://www.amazon.com/dp/B085NKYTFW?psc=1&ref=ppx_yo2ov_dt_b_product_details

**Reply by socelillyhammer** [reply]
@mdehrlich is your remote working for the Arris VMS1100? is that the IPC4100?  I am having remote issues i.e. remote not working.

**Comment by GerryDazoo** [comment]
Make sure you have the proper VideoSource configured in your config.ini. That’s the number one reason for a blank screen.

**Comment by stansoltz** [comment]
can someone provide me a simple bullet point on how to get both my slings running?  do I just create 2 config files and then launch the slinger server?

**Reply by theweatherguy** [reply]
### Using two slingboxes at the same time.
Much simpler than you'd think.

1. Rename the first config file that is working from **config.ini**   to  **config1.ini** 
2. Duplicate your config1.ini file and rename the duplicate to  **config2.ini**
3. Edit the **config2.ini** file—

>  change the password to the 2nd slingbox's secret admin password (line 3)

> Change the ipaddress to the 2nd slingbox's ip address  (line 9)  Get that from your router
> It's OK to leave the port the same (in my slingbox 350 it was 5201)   (line 10)

>  Change the **SERVER** port (line 48) of the config2.ini file from 8080  to 8082

4.  **Save** the changes to config2.ini file.

5. Put both the config1.ini and confi2.ini files in your home folder.   (delete any original config.ini file if it's still there)

**To start slinger server with two slingboxes**,  in Terminal, enter each command and hit return  

> `sudo sysctl -w net.inet.tcp.sendspace=5501072``  (Mac only)
>    _enter your computer admin password when prompted_

> `python3 slingbox_server.py config1.ini config2.ini`


You can now access BOTH slingboxes _at the same time_ using VLC on two different iPads or Macs

for Slingbox 1     **http://192.168.2.168:8080/slingbox**    (substitute the IP address for your Mac running Slinger) 

for Slingbox 2    **http://192.168.2.168:8082/slingbox**     (substitute the IP address for your Mac running Slinger) 

The remote codes commands: 
`http://192.168.2.168:8080/Remote`
`http://192.168.2.168:8082/Remote`

**Reply by GerryDazoo** [reply]
Or.......
Use the unified config file feature documented in the latest release notes. One file two slingboxes.

**Reply by stansoltz** [reply]
thanks folks.  will try in the a.m.  👍

**Reply by theweatherguy** [reply]
I see that you've released version 3.08 with the unified config.  Very cool.  Thank you!    Is there any way for you to automatically inform people here that a new version has been released?

**Reply by GerryDazoo** [reply]
Yes. In announcements. From OCt 16th
https://github.com/GerryDazoo/Slinger/discussions/136

**Comment by GerryDazoo** [comment]
Yes, two configs. Then pass both their names in on the command line. slingbox_server config1.ini config2.ini.

**Reply by stansoltz** [reply]
thanks folks.  will try in the a.m.  👍

**Comment by flipper2222222** [comment]
hi all, does this section 
[SLINGBOXES]
sb1=500
sb2=Solo
sb3=M1

get inserted between 

[SERVER]

and 

[REMOTE]

sections? thanks

**Reply by GerryDazoo** [reply]
It can go anywhere but I put it at the top I think it's clearer there. 
There is no [REMOTE] section in the unified config.
Check the release notes for an example.

**Reply by stansoltz** [reply]
Can someone post their config file with two slingboxes?  Just want to see how you stack the different boxes?

**Reply by theweatherguy** [reply]
Here's my unified_config file. (obscured my passwords for privacy)

; This a sample of a unified config.ini for three slingboxes

[SLINGBOXES]
sb1=Den
sb2=Carol
;sb3=M1

[Den]
sbtype="350/500/M1"
password=569TsYINoTREASLPASSWORDoerb
;finderid=nogood4CF70531865D73650A16A0A536
ipaddress=192.168.2.146
port=5201
Resolution=12
FrameRate=60
VideoBandwidth=2000
VideoSmoothness=63
VideoSource=1
maxstreams=3
StartChannel=245
;RemoteLock=yes
Remote=Den_remote.txt

[Carol]
sbtype="350/500/M1"
password=ErsZ1NOTREAOLPASSORDTer
ipaddress=192.168.2.132
;finderid=BAD95DAED609364C8204F2BC5FBDC363
port=5201
Resolution=12
FrameRate=30
VideoBandwidth=2000
VideoSmoothness=63
VideoSource=1
;RemoteLock=yes
;StartChannel=155
Remote=Carol_remote.txt

;[M1]
;sbtype="350/500/M1"
;password=SOmePassword
;finderid=E706A7405882C03A076502246DAD632F
;ipaddress=192.168.117.129
;port=5301
;Resolution=12
;FrameRate=30
;VideoBandwidth=2000
;VideoSmoothness=63
;VideoSource=1
;Remote=m1_remote.txt

[SERVER]
; local port number for the server to listen on for connections
; Send all streaming and Remote request to this port number.
port=8080
maxremotestreams=3
URLbase=slingbox
enableremote=yes

**Reply by flipper2222222** [reply]
add another big thanks Gerry, thanks to you too weather guy, sample was helpful

**Reply by theweatherguy** [reply]
Ooops.. I made one mistake in posting this copy of the config file.  The remote name for the first box should be  **Remote=Den_remote.txt**, not 350_remote.txt.

I've corrected it above too.   T**he name of each remote needs to be the same as the name of you give to each slingbox**  (like "Den" in this example).  Gerry has also said to avoid using numbers in the names you assign to your slingboxes

**Reply by GerryDazoo** [reply]
Two things.
1) I misspoke when I said don't numbers (I meant just some thing that could be interpreted as an integer). But this is not a restriction. See config file below
2) The remote file name can be anything. It does not have to match the name. I have multiple SB with the same remote file. What I think you might be referring to is my own personal naming convention for remote files.

slingbox 500 and 500lbr both use the same remote file.

[SLINGBOXES]
sb1=500
sb2=Solo
sb3=M1
sb4=500lbr

[500]
sbtype="350/500/M1"
password=lxk0VFQ8CY8j8sx
ipaddress=192.168.117.110
port=5201
Resolution=12
FrameRate=60
VideoBandwidth=2000
VideoSmoothness=63
VideoSource=1
maxstreams=3
;StartChannel=155
;StartChannel=35+9+13+13+42+42
;StartChannel=4.1
;RemoteLock=yes
Remote=500_remote.txt
;Remote=Remote.html

[Solo]
sbtype="Solo/Pro/ProHD"
password=12345     
ipaddress=192.168.117.101
port=5001
Resolution=12
FrameRate=30
VideoBandwidth=2000
VideoSmoothness=63
VideoSource=2
RemoteLock=yes
;StartChannel=155
Remote=solo_remote.txt

[M1]
sbtype="350/500/M1"
password=3s4z2141a121ujn
;finderid=E706A7405882C03A076502246DAD632F     
ipaddress=192.168.117.129
port=5301
Resolution=12
FrameRate=30
VideoBandwidth=2000
VideoSmoothness=63
VideoSource=1
Remote=m1_remote.txt

[500lbr]
sbtype="350/500/M1"
password=nC4tc2XfIalAak
ipaddress=192.168.117.110
port=5201
; Following are the default values
; Default resolution when server starts
Resolution=5

; Valid Values 1, 6, 10, 15, 20, 30, 60   depending on resolution
FrameRate=60

;Min 50 Max 8000
VideoBandwidth=1000
VideoSource=1

; Min 0 (auto) max 63
;VideoSmoothness=63

; Send Iframe every n seconds Max 30
;IframeRate=5

; Audio Bit Rate. Valid Options 16, 20, 32, 40, 48, 64, 96 Default 32
AudioBitRate=96
Remote=500_remote.txt

[SERVER]
; local port number for the server to listen on for connections
; port number+1 is used by the remote control code.
port=65432
maxremotestreams=1
URLbase=Gerry
;enableremote=no

**Reply by theweatherguy** [reply]
OK.  Thank you for clarifying this.  And thanks for posting your config file! 

That said, I imagine each remote needs a unique/different name so that the channel doesn't change on all the slingboxes, just the one you want to change the channel on. Does it makes sense to name them the same as each slingbox name to make the URLs for the remotes easier to remember?

(I think on version 3.07, you were using the URL of each slingbox for URL uniqueness; now you're using the  remote name assigned for uniqueness(?).   Maybe I'm wrong about that(?)

**Reply by GerryDazoo** [reply]
The remote file name has nothing to do with anything it's just a name. It has no effect one what URL you use to access the server for a particular slingbox.
All it does is to specify the Keycodes to be sent to the slingbox when you press a button.
It's the slingbox that sends out the IR signal based on the remote type you configured when to setup your video source. So it's the slingbox that is in control.

**Reply by GerryDazoo** [reply]
SO I think you're missing something here. With the unified config file you access your sling box by name. The general pattern is http://SERVER_IP:SERVER_PORT/URLbase/slingbox_id
For for example in my sample file I'd use these 4 URL to access each of my configured slingboxs
http://192.168.117.5:65432/Gerry/500
http://192.168.117.5:65432/Gerry/Solo
http://192.168.117.5:65432/Gerry/M1
http://192.168.117.5:65432/Gerry/500lbr
to get to the remote replace URLbase with Remote
http://192.168.117.5:65432/Remote/500
http://192.168.117.5:65432/Remote/Solo
http://192.168.117.5:65432/Remote/M1
http://192.168.117.5:65432/Remote/500lbr

For your supplied config
it would be http://YourServerIP:8080/slingbox/Carol  and
http://YourServerIP:8080/slingbox/Den

**Reply by theweatherguy** [reply]
Ok.  I understand.   I posted some new Mac Step-By Step instructions and I think I constructed them correctly there.   I'll quote your pattern to make it clearer.  Thanks.

**Reply by theweatherguy** [reply]
Now I understand!  I was a bit slow with this.  Thanks Gerry.

**Comment by stansoltz** [comment]
Any idea why I can't get my two slingboxes working?
**Config file:**

[SLINGBOXES]
sb1=350
sb2=ProHD

[350]
sbtype=350
name=350
password=______________
finderid=__________________________
ipaddress=192.168.1.145
port=5207
Resolution=12
FrameRate=30
VideoBandwidth=2000
VideoSmoothness=63
VideoSource=1
Remote=remote.txt

[ProHD]
sbtype=ProHD
name=ProHD
password=__________
finderid=______________________
ipaddress=192.168.1.254
port=8080
Resolution=12
FrameRate=30
VideoBandwidth=2000
VideoSmoothness=63
VideoSource=3
Remote=remote.txt

[SERVER]
port=55569
maxstreams=10
URLbase=pinehurst


**And when I try to reach either I get:**

Version : 3.08c Running on Windows-10-10.0.19043-SP0 pid= 7916
Using config file config.ini
Connection Manager Running on port 55569 with 10 max streams using URL pinehurst.
BOXES [('sb1', '350'), ('sb2', 'ProHD')]
Streamer Running:  10 config.ini ProHD ProHD 55569 8388608
starting up on  port 55569 8388608
Streamer Running:  10 config.ini 350 350 55569 8388608
Checking for slingbox at ('192.168.1.254', 8080)
Checking for slingbox at ('192.168.1.145', 5207)
('192.168.1.145', 5207) OK
350 Using slingbox at  ('192.168.1.145', 5207)
Streamer:  350 Waiting for first stream, flushing any IR requests that arrive while not connected to slingbox
('192.168.1.254', 8080) OK
ProHD Using slingbox at  ('192.168.1.254', 8080)
Streamer:  ProHD Waiting for first stream, flushing any IR requests that arrive while not connected to slingbox
Building page for 350
Reading Custom Remote definition from remote.txt
Building page for ProHD
Reading Custom Remote definition from remote.txt
 * Serving Flask app 'slingbox_server' (lazy loading)
 * Environment: production
   WARNING: This is a development server. Do not use it in a production deployment.
   Use a production WSGI server instead.
 * Debug mode: off
11/07/2022, 15:06:00  Registering Slingboxes {'__________________4': 55569, '1_______________________FF7': 55569}
Getting external IP address
Redirector, my external IP address 99.___.____.____.____
Registering _______________________4:55569
________________________FF7:55569

Register Successful
11/07/2022, 15:09:12   Streaming connection from ('192.168.1.1', 9995)

Slinginfo  ProHD 12 30 192.168.1.254 8080 3072 10
ProHD Starting Stream for  192.168.1.1:9995
Connecting... ('192.168.1.254', 8080) Control
ProHD Selecting Video Source 3
Keycodes= [1, 4, 5, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 21, 22, 24, 25, 27, 28, 29, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 46, 72, 73, 74, 75, 76, 77]
VideoParameters: Resolution= 12 FrameRate= 30 VideoBandwidth= 2000 VideoSmoothness= 63 IframeRate= 5 AudioBitRate= 64
Connecting... ('192.168.1.254', 8080) Stream
ProHD Stream started at 11/07/2022, 15:09:26  733 0
11/07/2022, 15:09:26  ProHD Stream Terminated for  192.168.1.1:9995 [WinError 10053] An established connection was aborted by the software in your host machine
ProHD Shutting down connections
Streamer:  ProHD Waiting for first stream, flushing any IR requests that arrive while not connected to slingbox
11/07/2022, 15:10:07   Streaming connection from ('192.168.1.1', 10243)

Slinginfo  350 12 30 192.168.1.145 5207 3072 10
350 Starting Stream for  192.168.1.1:10243
Connecting... ('192.168.1.145', 5207) Control
350 Selecting Video Source 1
Keycodes= [1, 4, 5, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 21, 22, 24, 25, 27, 28, 29, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 46, 72, 73, 74, 75, 76, 77]
VideoParameters: Resolution= 12 FrameRate= 30 VideoBandwidth= 2000 VideoSmoothness= 63 IframeRate= 5 AudioBitRate= 64
Connecting... ('192.168.1.145', 5207) Stream
350 Stream started at 11/07/2022, 15:10:10  744 0
11/07/2022, 15:10:10  350 Stream Terminated for  192.168.1.1:10243 [WinError 10053] An established connection was aborted by the software in your host machine
350 Shutting down connections
Streamer:  350 Waiting for first stream, flushing any IR requests that arrive while not connected to slingbox

**Comment by GerryDazoo** [comment]
Your sbtypes need to be "Solo/Pro/ProHD"
and "350/500/M1"

I'll take a look at supporting what you did.

**Reply by stansoltz** [reply]
I changed those to Solo/Pro/ProHD and 350/500/M1 and when I try to start the streams, I still get:

Register Successful
11/07/2022, 16:02:10   Streaming connection from ('192.168.1.1', 10557)

Slinginfo  Solo/Pro/ProHD 12 30 192.168.1.254 8080 3000 10
ProHD Starting Stream for  192.168.1.1:10557
Connecting... ('192.168.1.254', 8080) Control
ProHD Selecting Video Source 3
Keycodes= [1, 4, 5, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 21, 22, 24, 25, 27, 28, 29, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 46, 72, 73, 74, 75, 76, 77]
VideoParameters: Resolution= 12 FrameRate= 30 VideoBandwidth= 2000 VideoSmoothness= 63 IframeRate= 5 AudioBitRate= 64
Connecting... ('192.168.1.254', 8080) Stream
ProHD Stream started at 11/07/2022, 16:02:17  733 0
**11/07/2022, 16:02:17  ProHD Stream Terminated for  192.168.1.1:10557 [WinError 10053] An established connection was aborted by the software in your host machine**
ProHD Shutting down connections
Streamer:  ProHD Waiting for first stream, flushing any IR requests that arrive while not connected to slingbox
11/07/2022, 16:02:54   Streaming connection from ('192.168.1.1', 10680)

Slinginfo  350/500/M1 12 30 192.168.1.145 5207 3072 10
350 Starting Stream for  192.168.1.1:10680
Connecting... ('192.168.1.145', 5207) Control
350 Selecting Video Source 1
Keycodes= [1, 4, 5, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 21, 22, 24, 25, 27, 28, 29, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 46, 72, 73, 74, 75, 76, 77]
VideoParameters: Resolution= 12 FrameRate= 30 VideoBandwidth= 2000 VideoSmoothness= 63 IframeRate= 5 AudioBitRate= 64
Connecting... ('192.168.1.145', 5207) Stream
350 Stream started at 11/07/2022, 16:02:57  744 0
**11/07/2022, 16:02:57  350 Stream Terminated for  192.168.1.1:10680 [WinError 10053] An established connection was aborted by the software in your host machine**
350 Shutting down connections
Streamer:  350 Waiting for first stream, flushing any IR requests that arrive while not connected to slingbox

**Reply by GerryDazoo** [reply]
What are you using to view the video? 
Here's a simple test.  Point your browser at http://127.0.0.1:55569/pinehust/350
It should connect and start downloading the video stream ( which you can watch later if you want)
If the stream says up it's something not working with your media player.

**Reply by theweatherguy** [reply]
Just a suggestion to troubleshoot.  I would _simplify_ your setup and then make it fancy _after_ it works-

1.  I would eliminate the custom remote use here and just use the default remote settings in the config.ini file  to start
2.  I think you're supposed to keep the **finderid commented out** (;)   (?)
3.  And I would consider using the default port 8080 for the server just to simplify 
4.  I would run the command   ` netstat -a` in the  Windows terminal  (without the slinger server running) to check that the port #s you've chosen for Slingbox_server.py config.ini aren't already in use by other programs.  You may have hit the port lottery with an inadvertent conflict. 

I'm just trying to help.  (I've had to eliminate using the Fios Fancy Remote.txt for now.   What had been working fine stopped working for one of my boxes.  Not sure why.  Went back to the default remote.txt and all is fine.  Will experiment again in the future. )

**Reply by GerryDazoo** [reply]
TWG:
1. The custom remote will have no effect on the video stream.
2. In this case it's properly configured and working "Register successful". So I'm OK with that
4: That's very unlikely, as you say. But he can start a stream so the chosen port number is OK.

**Reply by theweatherguy** [reply]
Ok.   You're the expert on this, for sure. 

I guess I'll have to figure out why my FIOS_fancy_text remote.txt made my second slingbox inaccessible.    Removed it and went back to default.txt and it works.  Tried FIOS_Fancy.txt again...it was inaccessible...   Went back to default again....   it's OK. There was no other change in config.ini file.  

So I'm stumped if the custom remote.txt has no effect.  (Could something in the remote.txt code change the video source?  My two slingboxes have different video sources.) 

I see there's a new version of the FIOS remote txt.   I'll have to give that a try.

**Reply by GerryDazoo** [reply]
Please supply all the server output when you use the 'bad' fancy remote. Maybe there's something in there that's crashing the server.
There's no way for the remote to change the video source.

**Reply by stansoltz** [reply]
> What are you using to view the video? Here's a simple test. Point your browser at http://127.0.0.1:55569/pinehust/350 It should connect and start downloading the video stream ( which you can watch later if you want) If the stream says up it's something not working with your media player.

I'm using VLC.  I tried your test in the browser and both boxes started to download something...  but I couldn't figure out a way to stop the download, so I killed the browser via task manager and seemed to have lost the actual download file (hence I can't confirm if it was a video file or not, but based on the continuous download, I assume it was).

**Reply by GerryDazoo** [reply]
It's your firewall blocking external access.

**Reply by stansoltz** [reply]
> It's your firewall blocking external access.

Then why do I get a similar error if I try to access the server internally via my PC's DHCP address...  ie. 192.168.1.106:55569, or external WAN address?  ie. 99.224.xx.xxx:55569?  if it was a firewall, would the 192.168.1.106 not work?  Or, are you saying Windows firewall vs Router port-forwarding ?

EDIT.  I just shut off Windows Firewall, tried again locally via 192.168.1.106:55569 and same error.

**Reply by GerryDazoo** [reply]
Please supply the "exact" string your using. the server output, the output from the "ipconfig" command and a copy of the output of "netstat -an".  I'm thinking now you're not using the real PC IP address just the one you think it is.
